scipy/signal/_spline_filters.py

```diff
@@ -644,3 +647,161 @@ def qspline1d_eval(cj, newx, dx=1.0, x0=0):
         result += cj[indj] * _quadratic(newx - thisj)
     res[cond3] = result
     return res
+
+
+def symiirorder1(signal, c0, z1, precision=-1.0):
+    """
+    Implement a smoothing IIR filter with mirror-symmetric boundary conditions
+    using a cascade of first-order sections.
+
+    The second section uses a reversed sequence.  This implements a system with
+    the following transfer function and mirror-symmetric boundary conditions::
+
+                           c0
+           H(z) = ---------------------
+                   (1-z1/z) (1 - z1 z)
+
+    The resulting signal will have mirror symmetric boundary conditions
+    as well.
+
+    Parameters
+    ----------
+    input : ndarray
+        The input signal. If 2D, then the filter will be applied in a batched
+        fashion across the last axis.
```

Comment:
There's a naming mismatch here (`signal` in signature vs. `input` in docstring).